WooCommerce/Classes/ViewRelated/Dashboard/Factories/DashboardUIFactory.swift

```diff
@@ -0,0 +1,27 @@
+import UIKit
+
+protocol DashboardUI: UIViewController {
+    var refreshControl: UIRefreshControl { get }
+    func defaultAccountDidUpdate()
+    func reloadData(completion: @escaping () -> Void)
+}
+
+extension DashboardUI {
+    func displaySyncingErrorNotice() {
+        let title = NSLocalizedString("My store", comment: "My Store Notice Title for loading error")
+        let message = NSLocalizedString("Unable to load content", comment: "Load Action Failed")
+        let actionTitle = NSLocalizedString("Retry", comment: "Retry Action")
+        let notice = Notice(title: title, message: message, feedbackType: .error, actionTitle: actionTitle) { [weak self] in
+            self?.refreshControl.beginRefreshing()
+            self?.reloadData {}
+        }
+
+        AppDelegate.shared.noticePresenter.enqueue(notice: notice)
+    }
+}
+
+class DashboardUIFactory: NSObject {
```

Comment:
Is there something coming up that will require this class to conform to `NSObject`?

Reply:
no particular reason, just a habit from working with Objective-C 😅 will update and make the class `final`!